Dockerfile.build

```diff
@@ -1,5 +1,6 @@
 ARG GO_VERSION
-FROM docker.elastic.co/beats-dev/golang-crossbuild:${GO_VERSION}-main-debian11
```

Comment:
Speaking as someone seeing this for the first time, the fact that `ARCH` can be `main` is very surprising. Can we not use the built-in `TARGETARCH` and use `main` in the Dockerfile if the value is `amd64`?

Reply:
Or not use the built-in `TARGETARCH` as it seems misused in this case. Rename this ARG to `GO_ARCH` or another arbitrary name?

Reply:
The `main` arch seems a bit weird actually... 
@michel-laterman Does this come from some quirk on golang-crossbuild docker tags?

Reply:
yep, it's defined in the crossbuild readme

Reply:
> Or not use the built-in TARGETARCH arg at all, as it seems misused in this case. Use GO_ARCH (in line with GO_VERSION) or another arbitrary name?

I wanted to use TARGETARCH, because it lets us actually build a multiplatform image. But it looks like there isn't a way to conditionally use `main` in the crossbuild image tag based on this. It'd be best if the crossbuild image was multiplatform as well - then this would be completely transparent, and we'd just do `docker buildx build --platform=...` in the Makefile.

But as this cannot be done, I'm ok with the previous approach of using a custom variable for this.

Reply:
@v1v is there a specific reason why `main` is used instead of x86/amd64 ? I see the list of the images in [golang-crossbuild README](https://github.com/elastic/golang-crossbuild?tab=readme-ov-file#build-tags) and the `main` arch sticks out while all the other archs follow more conventional naming (arm, mips, ppc etc.)

But at least I can see now why `main` is used as default arg here, to target x86/amd64 architecture

Reply:
I believe `main` is a specific folder:
- https://github.com/elastic/golang-crossbuild/tree/main/go/main

See https://github.com/elastic/golang-crossbuild#gomain-docker-image

Reply:
I've changed the name variable name to `SUFFIX` to distance it from the `TARGETARCH/ARCH` concepts as it includes more information than just the architecture.
I also set it to use the same `main-debian11` suffix as our other targets use.

I'm more then happy to change to `TARGETARCH` if the upstream sources support it.